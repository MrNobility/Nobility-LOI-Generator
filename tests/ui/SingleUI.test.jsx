import { render } from '@testing-library/react';
import SingleUI from '../../src/ui/SingleUI';
import React from 'react'; // âœ… Required for JSX


test('SingleUI component matches snapshot', () => {
  const { container } = render(<SingleUI />);
  expect(container).toMatchSnapshot();
});
