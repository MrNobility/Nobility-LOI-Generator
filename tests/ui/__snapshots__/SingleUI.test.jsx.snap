// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleUI component matches snapshot 1`] = `
<div>
  <div
    class="bg-white rounded-xl shadow p-6 mb-8"
  >
    <h3
      class="text-xl font-semibold mb-4"
    >
      Single Deal
    </h3>
    <form
      class="space-y-4"
    >
      <div>
        <label
          class="block font-medium mb-1"
          for="single-offerType"
        >
          Offer Type
        </label>
        <select
          class="w-full border rounded p-2"
          id="single-offerType"
        >
          <option
            value="cash"
          >
            Cash Offer
          </option>
          <option
            value="sellerFinance"
          >
            Seller Finance Offer
          </option>
        </select>
      </div>
      <div>
        <label
          class="block font-medium mb-1"
          for="single-toneStyle"
        >
          Tone Style
        </label>
        <select
          class="w-full border rounded p-2"
          id="single-toneStyle"
        >
          <option
            value="professional"
          >
            professional
          </option>
        </select>
      </div>
      <div>
        <label
          class="block font-medium mb-1"
          for="single-tsvData"
        >
          Deal Data (TSV)
        </label>
        <textarea
          class="w-full border rounded p-2 font-mono"
          id="single-tsvData"
          placeholder="Paste single-row TSV here (headers on first line)"
          rows="8"
        />
      </div>
      <button
        class="bg-blue-600 text-white px-4 py-2 rounded hover:bg-blue-700"
        type="button"
      >
        Generate LOI
      </button>
    </form>
    <div
      class="mt-6"
      id="single-loiResult"
    />
  </div>
</div>
`;
